internal/graph/checkingresourcestream.go

```diff
@@ -476,6 +491,14 @@ func (crs *checkingResourceStream) runProcess(alwaysProcess bool) (bool, error)
 				Metadata:            metadata,
 				AfterResponseCursor: rai.reachableResult.AfterResponseCursor,
 			}
+		} else {
+			if rai.reachableResult.Metadata.DispatchCount > 0 {
```

Comment:
Can we not just always add without the if? My understanding is that this will not take a mutex...

Reply:
Since it takes a LOCK instruction, I'd still prefer to avoid that if possible